pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/utilities/ProtocolData.java

```diff
@@ -356,6 +361,20 @@ public float getStatsSampleRatio() {
         return statsSampleRatio;
     }
 
+    private void parseSecurityProperties() {
+        // obtain identity of the end-user -- mandatory only when impersonation is enabled
+        if (SecureLogin.isUserImpersonationEnabled()) {
+            this.user = getProperty("USER");
+        } else {
+            this.user = getOptionalProperty("USER");
```

Comment:
Do we need the user property if impersonation is not enabled ?

Reply:
the property is part of the InputData, so we fill it up anyways. Disabling impersonation disables running the request in the SecurityContext of the user, but user info might be important for some other things (like logging).